File: MLR-spring/src/main/java/com/mlr/controller/ContactController.java
package com.mlr.controller;

import com.mlr.dto.ContactRequestDto;
import com.mlr.service.EmailService;
import jakarta.validation.Valid;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.Map;

@RestController
@RequestMapping("/api/contact")
public class ContactController {

    @Autowired
    private EmailService emailService;

    @PostMapping
    public ResponseEntity<?> sendContactEmail(@Valid @RequestBody ContactRequestDto request) {
        try {
            emailService.sendContactMail(request);
            return ResponseEntity.ok(Map.of("message", "Email sent successfully"));
        } catch (Exception e) {
            return ResponseEntity.status(500).body(Map.of("message", "Failed to send email"));
        }
    }
}

File: MLR/ContactController.cs
using Microsoft.AspNetCore.Mvc;
using MLR.DTOs;
using MLR.Services;

namespace MLR.Controllers
{
    [ApiController]
    [Route("api/contact")]
    public class ContactController : ControllerBase
    {
        private readonly EmailService _emailService;

        public ContactController(EmailService emailService)
        {
            _emailService = emailService;
        }

        [HttpPost]
        public async Task<IActionResult> Send([FromBody] ContactRequestDto request)
        {
            if (request == null)
                return BadRequest("Invalid request");

            try
            {
                await _emailService.SendContactMail(request);

                return Ok(new
                {
                    message = "Message sent successfully"
                });
            }
            catch (Exception ex)
            {
                return StatusCode(500, new
                {
                    message = "Failed to send message",
                    error = ex.Message
                });
            }
        }
    }
}
